core/src/main/scala/org/http4s/rho/bits/RequestLike.scala

```diff
@@ -0,0 +1,62 @@
+package org.http4s.rho.bits
+
+import org.http4s.{AttributeMap, AuthedRequest, EntityBody, Headers, HttpVersion, Method, Request, Uri}
+
+case class RequestLikeOps[R[_[_]], F[_]](r: R[F]) extends AnyVal {
+  def toRequest(implicit rl: RequestLike[R]): Request[F] = rl.toRequest(r)
+  def method(implicit rl: RequestLike[R]): Method = rl.method(r)
+  def uri(implicit rl: RequestLike[R]): Uri = rl.uri(r)
+  def httpVersion(implicit rl: RequestLike[R]): HttpVersion = rl.httpVersion(r)
+  def headers(implicit rl: RequestLike[R]): Headers = rl.headers(r)
+  def body(implicit rl: RequestLike[R]): EntityBody[F] = rl.body(r)
+  def attributes(implicit rl: RequestLike[R]): AttributeMap = rl.attributes(r)
+  def pathInfo(implicit rl: RequestLike[R]): String = rl.pathInfo(r)
+}
+
+trait RequestLike[R[_[_]]] {
```

Comment:
This may be something that ties into @ChristopherDavenport's idea of a typeclass-based message.